cmd/api/http/handlers.go

```diff
@@ -421,3 +423,18 @@ func extractPageParams(query url.Values) (page, pageSize int, valid bool) {
 
 	return page, pageSize, true
 }
+
+func handleError(err error, w http.ResponseWriter, r *http.Request) {
+	if errors.Is(err, context.DeadlineExceeded) {
+		log.Println(err)
+		responseJSON(w, http.StatusGatewayTimeout, book.ErrResponseRequestTimeout)
+		return
+	}
+	if errors.Is(err, book.ErrResponseBookNotFound) {
+		log.Println(err)
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	log.Println(err)
+	w.WriteHeader(http.StatusInternalServerError)
+}
```

Comment:
You could use this function to avoid having to write this error handling in every http handler.

Reply:
The only method that was not using `handleError` was listBooks because of the default (unknow) case of it was a BadRequest. Now that we change it to an InternalServerError, like in the other methods, it fits `handleError` too.